src/lib/Metadata/Metadata.js

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright(c) ConsenSys Software Inc.
+ * Copyright(c) https://consensys.net/
+ * MIT Licensed
+ */
+
+import { readFileSync } from 'fs';
+
+import IPFS from '../../services/ipfsService.js';
+import { ERROR_LOG, errorLogger } from '../error/handler.js';
+
+export default class Metadata {
+  // TODO: implement logger
+  DEFAULT_MEDIA_ERROR = 'Image files has not been uploaded: ';
+
+  DEFAULT_METADATA_ERROR = 'Metadata file has not been uploaded: ';
+
+  VALIDATION_ERROR = 'Input should be a valid JSON or a filepath: ';
+
+  UNSUPPORTED_ERROR = 'Input is not supported (not JSON or filepath): ';
+
+  constructor({ projectId, projectSecret, ipfsUrl = 'ipfs.infura.io:5001' }) {
+    // TODO: change location for logger
+    if (!projectId) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.Ipfs_constructor,
+          message: ERROR_LOG.message.no_infura_projectID_supplied,
+        }),
+      );
+    }
+
+    // TODO: change location for logger
+    if (!projectSecret) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.Ipfs_constructor,
+          message: ERROR_LOG.message.no_infura_projectSecret_supplied,
+        }),
+      );
+    }
+
+    this.client = new IPFS({
+      ipfsUrl,
+      projectId,
+      projectSecret,
+    });
+  }
+
+  async createTokenURI(metadataInput) {
+    let result;
+    const metadata = this.parseInput(metadataInput);
+    try {
+      // upload 'image' property to IPFS
+      if (metadata.image) {
+        const cidImg = await this.client.uploadFile({ source: metadata.image });
+        metadata.image = `https://ipfs.io/ipfs/${cidImg}`;
+      }
+      // upload 'animation_url' property to IPFS
+      if (metadata.animation_url) {
+        const cidAnim = await this.client.uploadFile({ source: metadata.animation_url });
+        metadata.animation_url = `https://ipfs.io/ipfs/${cidAnim}`;
+      }
+    } catch (error) {
+      console.log(error); // eslint-disable-line no-console
+      throw (this.DEFAULT_MEDIA_ERROR, error);
+    }
+
+    try {
+      // upload metadata file to IPFS
+      const cidFile = await this.client.uploadObject({ source: metadata });
+      result = {
+        type: 'nftMetadata',
+        cid: cidFile,
+        ...metadata,
+      };
+    } catch (error) {
+      console.log(error); // eslint-disable-line no-console
+      throw new Error(this.DEFAULT_METADATA_ERROR, error);
+    }
+
+    return result;
+  }
+
+  async createContractURI(metadataInput) {
```

Comment:
We can dry up the code by either: putting the try/catch into their own functions and reusing inside of the `create...` methods or just have one create method with a `includeAnimation = false/true` flag

Reply:
I prefer 1

Reply:
Tried one other implementation. WDYT?

Reply:
@efecarranza is this implementation ok for you? BTW, did you checked the implementation example from Dayan? See here: https://gist.github.com/dayksx/81913e1552bba4fd51d537e592b002a7 (checked the comments)